src/daemon.js

```diff
@@ -27,15 +27,17 @@ const jsDaemon = {
 class Daemon {
   /**
    * @constructor
-   * @param {String} type daemon implementation type ("go" or "js")
+   * @param {String} spec daemon implementation spec (type "go" or "js")
    * @param {Multiaddr} addr multiaddr for the client to connect to
    * @param {Number} port port for the client to connect to
    */
-  constructor (type, addr, port) {
+  constructor (spec, addr, port) {
+    const {type, keyFile} = spec
```

Comment:
We shouldn't need to change the constructor since we can just pass the key path to the `start()` call.

Reply:
@jacobheun OK, will look at it later. Right now I'm working through some bugs triggered by the tests.